### ISSUE TITLE
[v0.4] Incorrect error on accessing ID on admin UI

### ISSUE THREAD

**Issue by vitalbone** [keystonejs/keystone#3092]
When trying to directly view an item by ID in the admin UI but ID does not exist, 3 incorrect behaviours occur.

1. Error message does not match the actual case: `item not found`
![screen shot 2016-06-30 at 9 54 25 am](https://cloud.githubusercontent.com/assets/5966259/16472702/7a4ac636-3ea9-11e6-8fd2-cc81e4de62bf.png)

2. Admin UI doesn't acknowledge that an error exists, therefore:

3. Will continue to make API requests resolving with a 404.
```
...
GET /keystone/api/locations/561f3ff4d2914aec21ce9ca1?drilldown=true 404 2.245 ms
GET /keystone/api/locations/561f3ff4d2914aec21ce9ca1?drilldown=true 404 1.392 ms
GET /keystone/api/locations/561f3ff4d2914aec21ce9ca1?drilldown=true 404 1.315 ms
GET /keystone/api/locations/561f3ff4d2914aec21ce9ca1?drilldown=true 404 1.428 ms
etc...
```

@mxstbr yo.

**Comment by mxstbr** [comment]
Fixed on latest `master` already, see #3059.

**Comment by vitalbone** [comment]
nice. one. mate.
